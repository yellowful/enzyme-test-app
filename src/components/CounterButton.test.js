import React from "react";
import {shallow} from 'enzyme';
import CounterButton from "./CounterButton";

// it('not clicked yet',()=>{
//     const shallowWrapper = shallow(<CounterButton />);
//     expect(shallowWrapper.state().count).toEqual(1);
// })

it('state test',()=>{
    const shallowWrapper = shallow(<CounterButton />);
    expect(shallowWrapper).toMatchSnapshot();
    shallowWrapper.find('button').simulate('click');
    expect(shallowWrapper.state().count).toBe(2);
    shallowWrapper.find('button').simulate('click');
    expect(shallowWrapper.state().count).toBe(3);
})

// it('state test',()=>{
//     const mountWrapper = mount(<CounterButton />);
//     mountWrapper.setState({ count:4 });
//     expect(mountWrapper.text()).toContain(4);
//     expect(mountWrapper).toMatchSnapshot();
// })

it('props test',()=>{
    
    const shallowWrapper = shallow(<CounterButton color='red' size='big' />);
    
    // console.log('props',shallowWrapper.props());
    expect(shallowWrapper.props().color).toBe('red');
    // 這裡的.props是指這個component return出來的element裡面的props，而不是<CounterButton />裡面的props
    expect(shallowWrapper.props().shape).toBe('big');
})