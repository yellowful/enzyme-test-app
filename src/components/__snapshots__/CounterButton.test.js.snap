// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state test 1`] = `
<button
  onClick={[Function]}
>
  Count: 
  1
</button>
`;
